students/allen_maxwell/Lesson_04/mailroom.py

```diff
@@ -0,0 +1,201 @@
+
+# Creates a single donor dictionary
+donors = {
+    'William Gates, III': [653784.49, 2, 326892.24],
+    'Mark Zuckerberg': [16396.10, 3, 5465.37],
+    'Jeff Bezos': [877.33, 1, 877.33],
+    'Paul Allen': [708.42, 3, 236.14],
+    'Steve Jobs': [10000.00, 1, 10000.00]}
+
+def menu():
+    '''
+    Creates a user input menu selection.
+     Input:  None
+     Output: None '''
+
+    # Loops through the menu options until the user selects 3, to quit
+    while True:
+        response = input('\n'.join(('\n',
+            'Please choose from the following options:',
+            '1) Create a Report',
+            '2) Send a Thank You letter to a single donor',
+            '3) Send a Thank You letter to all donors',
+            '4) Quit',
+            'Please enter your selection > ')))  
+        menu_func(response)
+
+def menu_func(selection):
+    '''
+    Processes menu input using dictionary values for switching.
+     Input:  String: numeric value
+     Output: None'''
+
+    menu_dict = {
+        1: print_report,
+        2: thank_you,
+        3: thank_all,
+        4: quit,}
+    
+    # Check if selection is a valid input
+    if not selection.isnumeric() or int(selection) not in menu_dict:
+        print('\nPlease enter a number 1 - 4')
+    else:
+        # Gets and calls the value associated with the selection
+        menu_dict.get(int(selection))()
+
+def sort_key(values): 
+    '''
+    Sorts the dictionary by values.
+     Input:  Dictionary: values
+     Output: Dictionary: sorted values'''
+    return values[1][0]
+
+def print_report():
+    '''
+    Prints a sorted list of your donors, sorted by total historical donation amount.
+     Input:  None
+     Output: Display: formatted donor list'''
+
+    # Create Report Header
+    print('\n{:<19}{:3}{:12}{:3}{:10}{:3}{:<10}'
+    .format('Donor Name', '|', 'Total Given', '|', 'Num Gifts', '|', 'Average Gift'))
+    print('-'*62)
+
+    # Print donor list into a table format
+    for name, donor in sorted(donors.items(), key = sort_key, reverse = True):
+        print('{:<21}${:>11,.2f}{:^16}${:>11,.2f}'.format(name, *donor))
+
+def thank_you():  
+    '''
+    Creates a Thank You letter for a donor.
+     Input:  None
+     Output: None'''
+
+    found = False
+    while not found:
+
+        # Prompt for a full name of the donor, list of donor names, or quit to menu
+        response = input("\nPlease enter the full name of the donor, 'List' for a list of donors," 
+            " or 'Quit' to exit to the menu > ")
+
+        # Exit to main menu if 'quit' is entered
+        if response.lower() == 'quit':
+            menu()
+
+        # If input value is empty
+        if response == '':
+            print('\nPlease enter a name')
+        # If response is list show a list of donor names and reprompt
+        elif response.lower() == 'list':
+            for donor in sorted(donors):
+                print(donor)    
+        else:
+            for donor in donors:        
```

Comment:
you shouldn't loop through donors dictionary, if you would like to check if donor exists then utilize the efficiency of dictionary -> `donors.get(response)`